doc/source/cluster/kubernetes/user-guides/rayservice.md

```diff
@@ -0,0 +1,299 @@
+
+## Step 7: In-place update for Ray Serve applications
+
+You can update the configurations for the applications by modifying `serveConfigV2` in the RayService config file. Re-applying the modified config with `kubectl apply` will re-apply the new configurations to the existing RayCluster instead of creating a new RayCluster.
+
+Let's try it out. Update the price of mangos from `3` to `4` for the fruit stand app in [ray_v1alpha1_rayservice.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml). This will reconfigure the existing MangoStand deployment, and future requests will use the updated Mango price.
+
+```sh
+# Step 7.1: Update the price of mangos from 3 to 4.
+# [ray_v1alpha1_rayservice.yaml]
+# - name: MangoStand
+#   num_replicas: 1
+#   user_config:
+#     price: 4
+
+# Step 7.2: Apply the updated RayService config.
+kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+# Step 7.3: Check the status of the RayService.
+kubectl describe rayservices rayservice-sample
+# [Example output]
+# Serve Deployment Statuses:
+# - healthLastUpdateTime: "2023-07-11T23:50:13Z"
+#   lastUpdateTime: "2023-07-11T23:50:13Z"
+#   name: MangoStand
+#   status: UPDATING
+
+# Step 7.4: Send a request to the fruit stand app again after the Serve deployment status changes from UPDATING to HEALTHY.
+# (Execute the command in the curl Pod from Step 6)
+curl -X POST -H 'Content-Type: application/json' rayservice-sample-serve-svc:8000/fruit/ -d '["MANGO", 2]'
+# [Expected output]: 8
+```
+
+## Step 8: Zero downtime upgrade for Ray clusters
+
+In Step 7, modifying `serveConfigV2` will not trigger a zero downtime upgrade for Ray clusters.
+Instead, it will reapply the new configurations to the existing RayCluster.
+However, if you modify `spec.rayClusterConfig` in the RayService YAML file, it will trigger a zero downtime upgrade for Ray clusters.
+To elaborate, RayService will temporarily create a new RayCluster and wait for it to be ready, then switch traffic to the new RayCluster by updating the selector of the head service managed by RayService (i.e. `rayservice-sample-head-svc`) and terminate the old one.
+
+To elaborate, during the zero downtime upgrade process, RayService will create a new RayCluster temporarily and wait for it to become ready. 
+Once the new RayCluster is ready, RayService will update the selector of the head service managed by RayService (i.e., `rayservice-sample-head-svc`) to point to the new RayCluster to switch the traffic to the new RayCluster. 
+Finally, the old RayCluster will be terminated.
+
+There are certain exceptions that will not trigger a zero downtime upgrade.
+Currently, only the fields managed by Ray Autoscaler, such as `replicas` and `scaleStrategy.workersToDelete`, will not trigger a zero downtime upgrade.
```

Comment:
I think I might have asked before, but I think this could be made more precise.  

> Currently, only the fields managed by Ray Autoscaler, such as `replicas` and `scaleStrategy.workersToDelete`, will not trigger a zero downtime upgrade.

Because of "such as", this implies that there are also other fields that don't trigger it, and `replicas` and `workersToDelete` are just two examples.   If these are actually the only fields that don't trigger it, consider 

> Currently, only the fields `replicas` and `scaleStrategy.workersToDelete` will not trigger a zero downtime upgrade.  This is because these fields are managed by the Ray Autoscaler.

(I'm not sure if these are the only fields, though)

Reply:
Update https://github.com/ray-project/ray/pull/39074/commits/7c39cd2d2e8b9f983995c573188a0582a94ceab9